convert.py

```diff
@@ -15,65 +15,114 @@
 import os
 import argparse
 
-from onnx import helper
+from onnx import helper, checker
 import paddle.fluid as fluid
 
-import ops
-from variables import paddle_variable_to_onnx_tensor
+import fluid_onnx.ops as ops
+from fluid_onnx.variables import paddle_variable_to_onnx_tensor
+from fluid_onnx.variables import PADDLE_TO_ONNX_DTYPE
 
 
-def convert(dirname):
+def parse_args():
+    # Read arguments: path to model.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--fluid_model", required=True, help="Input PaddlePaddle Fluid model.")
+    parser.add_argument(
+        "--onnx_model_dir",
+        required=False,
+        help="The output directory for ONNX model.")
+    args = parser.parse_args()
+    return args
+
+
+def convert(args):
     # Read the model files.
     place = fluid.CPUPlace()
     exe = fluid.Executor(place)
 
     inference_scope = fluid.core.Scope()
     with fluid.scope_guard(inference_scope):
         [inference_program, feed_target_names,
-         fetch_targets] = fluid.io.load_inference_model(dirname, exe)
+         fetch_targets] = fluid.io.load_inference_model(args.fluid_model, exe)
 
         # Using blocks in programs, create nodes using:
         onnx_nodes = []
-        all_inputs = []
-        for block in inference_program.blocks:
-            all_inputs += [
-                paddle_variable_to_onnx_tensor(v, block) for v in block.vars
-                if v not in ['feed', 'fetch']
-            ]
 
+        # Load parameters
+        global_block = inference_program.global_block()
+        for var_name in global_block.vars:
+            var = global_block.var(var_name)
+            if var_name not in ['feed', 'fetch'] and var.persistable:
+                param = fluid.executor.fetch_var(var_name, inference_scope)
+                param_node = helper.make_node(
```

Comment:
So where did you figure out that this was the way params need to be set in an ONNX model? I have been confused and asking around

Reply:
Oh interesting. I asked this question on the ONNX gitter chat and searched a lot through their issues. So (a) your solution is good based on the info you have. (b) it might be a good idea to use what the practice they are recommending. Which is use the `initializers` argument on the `make_model` function. More here: https://github.com/onnx/onnxmltools/tree/master/onnxmltools/convert/common and https://github.com/onnx/onnxmltools/blob/master/onnxmltools/convert/coreml/NeuralNetwork/fullyconnected.py#L33. So basically collect the values while traversing the variables, and then when calling `make_model`, pass them in.

Reply:
This is an important information. I think that we can keep the current implementation and move on. In future we can try the ```initializers``` to see if it  works.

Reply:
I don't think so, `parameter` is a common concept, so we should make this clear.